lib/matplotlib/axes/_axes.py

```diff
@@ -5724,7 +5724,6 @@ def _interp_grid(X):
                 if nrows == Ny:
                     X = _interp_grid(X.T).T
                     Y = _interp_grid(Y.T).T
-                shading = 'flat'
 
```

Comment:
Maybe I missed it, but why can this be removed?

Reply:
When shading is set to auto and like mentioned above,
auto: Chooses nearest if x [M], y[N], or flat if x [M+1], y [N+1].
Shading is set back to being flat when x[M[, y[N] and C is of the same shape, this basically causes Quadmesh._shading to be set to flat regardless of the shading set to auto, it's never nearest when set to auto.

Reply:
Ping @jklymak Since you added that part: Is it ok to remove that here?

Reply:
Maybe we should get #20268 in first since that is the way this *should* work.

Reply:
shading can only be flat or gouraud.  `'nearest'` and `'auto'` create a mesh that is `'flat'` .  I think this statement needs to stay.

Reply:
Okay, so how do I go about checking the dimensions then?
`self.shading` will always be set to `'flat'` and and when shading is `'flat'` we can either accept 
x [M+1], y[N+1] or x[M], y[N] but for `'nearest'` we need x[M], y[N] only right.

Reply:
Probably best to rebase on master because #20268 is in, and flat with N, M now raises...

Reply:
> Probably best to rebase on master because #20268 is in, and flat with N, M now raises...

If that's the case when shading is set to `'auto'` `self.shading` is going to set to `'flat'` right? 
How would we validate those dimensions because `'auto'` when x [M], y[N] is correct but we can't tell because `self.shading` is always set to `'flat'`

Reply:
I think this is all explained in #20268?  If its not clear let me know...